### ISSUE TITLE
[XCUITest] Error: Unable to launch WebDriverAgent because of xcodebuild failure: "xcodebuild failed with code 65".

### ISSUE THREAD

**Issue by vikramvi** [appium/appium#9035]
## The problem

1. I've closely followed 2 tutorials namely
https://github.com/appium/appium/blob/master/docs/en/appium-setup/real-devices-ios.md
https://github.com/appium/appium-xcuitest-driver/blob/master/docs/real-device-config.md

2. Configuring iOS Real Device Tests: Xcode 8 and iOS 9.3 and above > Basic (automatic) configuration per this section followed all the steps

3. But even with this appium can only install app on the real device but fails to install WDA

4. Some confusing part of this READM is 
"Full manual configuration > Alternatively, the provisioning profile can be manually associated with the project (keep in mind that this will have to be done each time the WebDriverAgent is updated, and is not recommended):"
 In this why to do this ?

5. .ipa file is created with by choosing Team value as actual company name and not (Personal Team ), does this make any difference ?

## Environment

* Appium version (or git revision) that exhibits the issue: 1.6.6-beta.3
* Last Appium version that did not exhibit the issue (if applicable):
* Desktop OS/version used to run Appium: Mac OS  + 10.12.5
* Node.js version (unless using Appium.app|exe): v7.9.0
* Mobile platform/version under test: iPhone 4S + 9.3.5
* Real device or emulator/simulator: Real Device
* Appium CLI or Appium.app|exe: CLI

## Details

NA
## Link to Appium logs

[appium log](https://gist.github.com/vikramvi/a2e6a128510fb77b0ae7228eee0dfe9b) 

## Code To Reproduce Issue [ Good To Have ]

NA

**Comment by mykola-mokhnach** [comment]
> [Xcode] 2017-08-16 12:52:29.803 XCTRunner[188:3748] The bundle “WebDriverAgentRunner” couldn’t be loaded because it doesn’t contain a version for the current architecture. Try installing a universal version of the bundle.


You are trying to execute WDA on iPhone 4S, which does not support 64bit architecture. Try some newer device, for example iPhone 6

**Comment by vikramvi** [comment]
@mykola-mokhnach  Is there any workaround to run automation on 4S as our app needs to be tested with 9.x iOS version as well.

**Comment by mykola-mokhnach** [comment]
Nothing that I know about, at least with XCTest

**Comment by vikramvi** [comment]
As per section **Full manual configuration** under https://github.com/appium/appium-xcuitest-driver/blob/master/docs/real-device-config.md

I did the configuration and ran below command

`    xcodebuild -project WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -destination 'id=<udid>' test`

output is as below ( with WDA installed successfully on iPhone 4S )

```
vikram-anna-pro:WebDriverAgent vikram-anna$ xcodebuild -project WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -destination 'id=6930e9e68d8fe637f29a39768f1b846a72805a96' test
2017-08-16 14:51:34.544 xcodebuild[21581:228103]  IDETestOperationsObserverDebug: Writing diagnostic log for test session to:
/Users/vikram-anna/Library/Developer/Xcode/DerivedData/WebDriverAgent-brdadhpuduowllgivnnvuygpwhzy/Logs/Test/956DE5EA-78A7-4C94-8036-81B8CE90A54B/Session-WebDriverAgentRunner-2017-08-16_145134-XRlIAl.log
2017-08-16 14:51:34.544 xcodebuild[21581:228079] [MT] IDETestOperationsObserverDebug: (45609E36-3D7D-4BDC-BC3D-AB7E01928CC7) Beginning test session WebDriverAgentRunner-45609E36-3D7D-4BDC-BC3D-AB7E01928CC7 at 2017-08-16 14:51:34.544 with Xcode 8E3004b on target <DVTiOSDevice: 0x7fbfb410ef90> {
		deviceSerialNumber:         DX5LFEVSFML6
		identifier:                 6930e9e68d8fe637f29a39768f1b846a72805a96
		deviceClass:                iPhone
		deviceName:                 Chimmi iPhone
		deviceIdentifier:           6930e9e68d8fe637f29a39768f1b846a72805a96
		productVersion:             9.3.5
		buildVersion:               13G36
		deviceSoftwareVersion:      9.3.5 (13G36)
		deviceArchitecture:         armv7f
		deviceTotalCapacity:        5660655616
		deviceAvailableCapacity:    3252559872
		deviceIsTransient:          NO
		ignored:                    NO
		deviceIsBusy:               NO
		deviceIsActivated:          YES
		deviceActivationState:      Activated
		isPasscodeLocked:           NO
		deviceType:                 <DVTDeviceType:0x7fbfb6a40a20 Xcode.DeviceType.iPhone>
		supportedDeviceFamilies:    (
    1
)
		applications:              (null)
		provisioningProfiles:      (null)
		activityProgress:          -2
		activityTitle:             
		hasInternalSupport:        NO
		isSupportedOS:             YES
		developerDiskMountError:   (null)
(null)
	bootArgs:                  <unavailable>
		} (9.3.5 (13G36))
=== BUILD TARGET WebDriverAgentLib OF PROJECT WebDriverAgent WITH CONFIGURATION Debug ===

Check dependencies

=== BUILD TARGET WebDriverAgentRunner OF PROJECT WebDriverAgent WITH CONFIGURATION Debug ===

Check dependencies

writeDictToFile:1278 ==== Successfully wrote Manifest cache to /var/folders/xk/ryk32cv91t700bvnr4z3f5fw0000gn/C/com.apple.DeveloperTools/All/Xcode/EmbeddedAppDeltas/5355a73a38b804ecc0dc96373a96b482/6930e9e68d8fe637f29a39768f1b846a72805a96/ManifestCache.plist
2017-08-16 14:51:45.245 XCTRunner[265:20910] Running tests...
2017-08-16 14:51:50.150 XCTRunner[265:20910] Continuing to run tests in the background with task ID 1
Test Suite 'All tests' started at 2017-08-16 2:51:51.421 PM
Test Suite 'WebDriverAgentRunner.xctest' started at 2017-08-16 2:51:51.436 PM
Test Suite 'UITestingUITests' started at 2017-08-16 2:51:51.438 PM
Test Case '-[UITestingUITests testRunner]' started.
    t =     0.01s     Start Test at 2017-08-16 2:51:51.452 PM
    t =     0.04s     Set Up
2017-08-16 14:51:51.510 XCTRunner[265:20910] Built at Aug 16 2017 14:45:31
2017-08-16 14:51:51.665 XCTRunner[265:20910] ServerURLHere->http://192.168.2.51:0<-ServerURLHere
```

This indicates WDA is fine in stand alone mode on iPhone 4S,  please clarify if I'm missing something here.

I feel it may be appium / appium-xcuitest-driver bug ?  Please review

**Comment by vikramvi** [comment]
Also one more observation

As per instructions under **"...you can try accessing the WebDriverAgent server status..."**

```
vikram-anna-pro:~ vikram-anna$ export DEVICE_URL='http://192.168.2.51:8100'
vikram-anna-pro:~ vikram-anna$ export JSON_HEADER='-H "Content-Type: application/json;charset=UTF-8, accept: application/json"'
vikram-anna-pro:~ vikram-anna$ curl -X GET $JSON_HEADER $DEVICE_URL/status
curl: (6) Could not resolve host: application
curl: (6) Could not resolve host: accept
curl: (6) Could not resolve host: application
curl: (7) Failed to connect to 192.168.2.51 port 8100: Connection refused
```

**Comment by vikramvi** [comment]
OK, after above step I could able to launch app on **iPhone 4S** successfully  :)

[appium server success log](https://gist.github.com/vikramvi/e657be8a73d2f348a4afb05eb66bd64e)

This proves there are things needs to be fixed

1. Appium server log giving wrong info

2. Documentation needs to be updated as well, it is still quite confusing ( without Full manual configuration ) I doubt if somebody can able to get the setup up & running.

**Comment by imurchie** [comment]
What information in the logs is wrong?

**Comment by vikramvi** [comment]
@imurchie https://gist.github.com/vikramvi/a2e6a128510fb77b0ae7228eee0dfe9b#file-gistfile1-txt-L113

**Comment by imurchie** [comment]
This line?
```
[Xcode] 2017-08-16 12:52:29.803 XCTRunner[188:3748] The bundle “WebDriverAgentRunner” couldn’t be loaded because it doesn’t contain a version for the current architecture. Try installing a universal version of the bundle.
```

That is from Apple. Appium is just printing it out.

**Comment by vikramvi** [comment]
@imurchie Ok, so I feel documentation needs to be updated. 
I struggled a lot to get it working on iPhone 4S.

- To get this working I had to try out **Full manual configuration**  section
- Can we discuss issues in documentation in this issue or file a new ticket ?

**Comment by imurchie** [comment]
In this ticket is fine.

Sometimes it is necessary to do the full manual configuration. That is why that is in the documentation.

**Comment by vikramvi** [comment]
@imurchie  Please find below observations wrt documentation

**Issue 1:**

There are multiple documents explaining this; I’m NOT sure why

1.1 https://github.com/appium/appium/blob/master/docs/en/appium-setup/real-devices-ios.md

1.2 https://github.com/appium/appium-xcuitest-driver/blob/master/docs/real-device-config.md

can we have just one copy please ?

**Issue 2**

2.1 Documentation should ask user to install app on the physical device successfully first with help of XCode and if step is success **then only** goto next stop of installing WDA.
2.2 Without app being ready to be installed on test devices, no point going to next step

**Issue 3**

3.1 "Basic (manual) configuration" section, not sure how & from where WebDriverAgent is opened. Need more info here
3.2 "by going into your account preferences and seeing the provisioning profile" this screen shot is not valid for XCode 8.3.3
3.3 updatedWDABundleId never worked for me, after step 3.2

**Issue 4**

4.1 "Full manual configuration" section "..be done each time the WebDriverAgent is updated" 
      does WebDriverAgent gets updated whenever new version of appium is installed ?

4.2  I found that Xcode > Team value should always be  **YOUR_COMPANY_NAME** ,  **YOUR_NAME ( Personal Team) ** never worked.


If you can give feedback here, I am happy to submit PR as well. It can be useful to many users

**Comment by vikramvi** [comment]
ping @imurchie

**Comment by vikramvi** [comment]
@imurchie  @jlipps  @mykola-mokhnach  what shall we do with this issue ? I had already proposed documentation changes couple of weeks back.

**Comment by jlipps** [comment]
@vikramvi those look like excellent improvements. Can you make the PR you proposed?

**Comment by vikramvi** [comment]
@jlipps sure I will create PR and we can discuss & clarify open questions in above comment

**Comment by jlipps** [comment]
hi @vikramvi just wanted to ping and ask if you are going to make the PR

**Comment by vikramvi** [comment]
@jlipps I will submit by next weekend. sorry for long delay

**Comment by Arav4prot** [comment]
Hi Guys,

I am getting the same issue. 

https://github.com/appium/appium/issues/11590

Please do the needful as I am unable to  move forward
